spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/JdbcAggregateTemplate.java

```diff
@@ -309,11 +313,43 @@ public void deleteAll(Class<?> domainType) {
 
 		entity = triggerBeforeDelete(entity, id, change);
 
-		executor.execute(change);
+		executor.executeDelete(change);
 
 		triggerAfterDelete(entity, id, change);
 	}
 
+	private <T> List<T> performSaveChange(Iterable<T> instances, Function<T, Function<T, AggregateChangeWithRoot<T>>> changeCreatorSelector) {
```

Comment:
`Function` of `Function` introduces quite some complexity. Could we simplify this to return a `AggregateChangeWithRoot<T>` instead `Function<T, AggregateChangeWithRoot<T>>`?